src/qibolab/instruments/qblox/controller.py

```diff
@@ -423,93 +420,84 @@ def _sweep_recursion(
+                navgs = options.nshots if options.averaging_mode != AveragingMode.SINGLESHOT else 1
+                num_bins = nshots
+                for sweeper in sweepers:
+                    num_bins *= len(sweeper.values)
 
                     # split the sweep if the number of bins is larget than the memory of the sequencer (2**17)
-                    if num_bins < 2**17:
-                        # for sweeper in sweepers:
-                        #     if sweeper.parameter is Parameter.amplitude:
-                        #         # qblox cannot sweep amplitude in real time, but sweeping gain is quivalent
-                        #         for pulse in sweeper.pulses:
-                        #             pulse.amplitude = 1
-
-                        #     elif sweeper.parameter is Parameter.gain:
-                        #         for pulse in sweeper.pulses:
-                        #             # qblox has an external and an internal gains
-                        #             # when sweeping the internal, set the external to 1
-                        #             # TODO check if it needs to be restored after execution
-                        #             if pulse.type == PulseType.READOUT:
-                        #                 qubits[pulse.qubit].readout.gain = 1
-                        #             elif pulse.type == PulseType.DRIVE:
-                        #                 qubits[pulse.qubit].drive.gain = 1
-
-                        result = self._execute_pulse_sequence(qubits, sequence, options, sweepers)
-                        for pulse in sequence.ro_pulses:
-                            if results[pulse.id]:
-                                results[pulse.id] += result[pulse.serial]
-                            else:
-                                results[pulse.id] = result[pulse.serial]
-                            results[pulse.qubit] = results[pulse.id]
+                if num_bins < 2**17:
+                    # for sweeper in sweepers:
+                    #     if sweeper.parameter is Parameter.amplitude:
+                    #         # qblox cannot sweep amplitude in real time, but sweeping gain is quivalent
+                    #         for pulse in sweeper.pulses:
+                    #             pulse.amplitude = 1
+
+                    #     elif sweeper.parameter is Parameter.gain:
+                    #         for pulse in sweeper.pulses:
+                    #             # qblox has an external and an internal gains
+                    #             # when sweeping the internal, set the external to 1
+                    #             # TODO check if it needs to be restored after execution
+                    #             if pulse.type == PulseType.READOUT:
+                    #                 qubits[pulse.qubit].readout.gain = 1
+                    #             elif pulse.type == PulseType.DRIVE:
+                    #                 qubits[pulse.qubit].drive.gain = 1
```

Comment:
I know that this was already commented but if it is not used I suggest to remove it.

Reply:
I would like to speak with Alvaro before removing this.

Reply:
I would also suggest to stick to one thing per PR so that it is easier to review and fix any introduced bugs. Since this PR is about the instrument settings, avoid touching the sweepers as much as possible, even if you see things that you don't like. If it's small fixes, fine, but if it's a big change it may introduce some bug which will be harder to solve the bigger the PR is (and the settings part is already big).